### PR TITLE
readme: update readme descriptions

### LINKED ISSUE #34824: Descriptions in readme are long and not friendly to quick start
## Enhancement

- Use links of official documentation to make [readme](https://github.com/pingcap/tidb/blob/master/README.md) concise and easier to quick start
- Update description style of the quick start section

### PR DESCRIPTION
**Pull request by lilin90**
### What problem does this PR solve?

Issue Number: close #34824

Update descriptions and content organization in readme

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,33 +14,29 @@
...
+For more details and latest updates, see [TiDB docs](https://docs.pingcap.com/tidb/stable) and [release notes](https://docs.pingcap.com/tidb/dev/release-notes).
 
-    TiDB internally shards table into small range-based chunks that we refer to as "Regions". Each Region defaults to approximately 100 MiB in size, and TiDB uses an [optimized](https://pingcap.com/blog/async-commit-the-accelerator-for-transaction-commit-in-tidb-5.0) Two-phase commit to ensure that Regions are maintained in a transactionally consistent way.
+## Quick start
 
-- __Cloud Native__
+### Start with TiDB Cloud
 
-    TiDB is designed to work in the cloud -- public, private, or hybrid -- making deployment, provisioning, operations, and maintenance simple.
+We can run TiDB for you, so you don't have to run your own cluster.
```
[the remark this answers is not in the document]
Comment (dcalvin) [reply README.md:27]: Who are we? It's too marketing here.

**Review by dcalvin — COMMENTED** [review]
Only one edit. Rest LGTM

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (lilin90) [reply README.md:27]: I directly edited the suggested change.

**Review by dcalvin — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Review by winoros — APPROVED** [review]

**Pull request merged**